Add type definitions for Jest

p-jackson opened the pull request:
#### Changes proposed in this Pull Request

This fixes the type errors test files related to `describe`, `it`, `expect`, etc. It also improves auto-complete.

Installed version 24.x.x so it lines up with version 24.x.x of Jest.

* Add `@types/jest` as a `devDependency`

#### Testing instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Open TypeScript test file (e.g. `client/lib/interval/test/interval.tsx`)
* See type errors for `describe` and `it`
* `npm install`
* Type errors gone

sirreal commented:
Thanks! We could consider adding this to `calypso-build` and adding it to the `types` tsconfig like in https://github.com/Automattic/wp-calypso/pull/38466:

https://github.com/Automattic/wp-calypso/blob/1ee3047d4eee44ab6c73ee3e239922e0db4c90af/packages/calypso-build/tsconfig.json#L19

(not landed at time of comment)

p-jackson commented:
> We could consider adding this to `calypso-build` and adding it to the `types` tsconfig

I like the idea of adding this to `calypso-build`, but when I gave it a try it didn't remove the errors in VS Code. I added the package to the `calypso-build/package.json` and then added `"jest"` to the array in `tsconfig.json`, is that what you were thinking?

p-jackson committed:
Add type definitions for Jest

This fixes the type errors in test files related to `describe`, `it`,
`expect`, etc. It also improves auto-complete.

p-jackson commented:
Rebased  with `master` after the node upgrade. The `package-lock.json` change is much tidier now :)

I'm going to merge this as is, it fixes my immediate problem, but there might be something nicer we can do with `calypso-build` in the future.
